r/tests/testthat/test-dataset-csv.R

```diff
@@ -253,7 +253,7 @@ test_that("readr parse options", {
       tsv_dir,
       partitioning = "part",
       format = "text",
-      quo = "\"",
+      del = ","
```

Comment:
Yep - because passing in a parameter as `quo` (not a valid parameter name) no longer constitutes a unique partial match to either the readr or arrow options, it is caught earlier on in input validation and so raises an "Unrecognized option" error instead of an "Ambiguous option" error.  This input validation is something it would be nice to refactor in another PR, but I figured here we just want to replace `quo` with something else that takes us down the "Ambiguous option" path to stick with the spirit of the test.